spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -670,6 +670,28 @@ def supports_value_expectations?
     end
   end
 
+  RSpec.describe 'custom matcher that does not specify if it supports value expectations' do
+    let(:value_matcher_class) do
+      Class.new do
+        def matches?(_actual); true; end
+        def failure_message_when_negated; "oops"; end
+      end
+    end
+    let(:value_matcher) { value_matcher_class.new }
+
+    before { expect(value_matcher).not_to respond_to(:supports_value_expectations?) }
+
+    it '`should` does not print a deprecation warning when given a value' do
+      expect_no_deprecations
+      expect { should value_matcher }.not_to raise_error
+    end
+
+    it '`should_not` does not print a deprecation warning when given a value' do
+      expect_no_deprecations
```

Comment:
I have far less clue than you do, but those sound like good hypotheses? I assumed we were handling Exception here (and not StandardError) because the expectation errors aren't descendants of the latter, but I hadn't given it any thought past that. I'm also guessing that the _reason_ for preferring `is_expected` over `should` is that the former can be implemented as a matcher, instead of in rspec-core?